File: rsa.js
const NodeRSA = require('node-rsa');

// Generate RSA key pair
const key = new NodeRSA({ b: 512 });

// Encrypt the data
const encrypted = key.encrypt('This is a secret message', 'base64');

// Decrypt the data
const decrypted = key.decrypt(encrypted, 'utf8');

console.log("Ciphertext:", encrypted);
console.log("Plaintext:", decrypted);

File: rsa.py
from Crypto.PublicKey import RSA
from Crypto.Cipher import PKCS1_OAEP

# Generate RSA key pair
key = RSA.generate(2048)
private_key = key.export_key()
public_key = key.publickey().export_key()

# Encrypt the data
cipher = PKCS1_OAEP.new(RSA.import_key(public_key))
ciphertext = cipher.encrypt(b'This is a secret message')

# Decrypt the data
cipher = PKCS1_OAEP.new(RSA.import_key(private_key))
plaintext = cipher.decrypt(ciphertext)

print("Ciphertext:", ciphertext)
print("Plaintext:", plaintext)
